### ISSUE TITLE
Feedback 05 - Availability

### ISSUE THREAD

**Issue by CDR-Register-Stream** [cdr-register/register#6]
This thread has been created to allow for feedback on the Availability. All feedback is welcome.

**Comment by NationalAustraliaBank** [comment]
NAB would like to provide the following initial feedback on Availability.

#### CDR Register Unavailable

- With the expectation to not share consumer data in specific scenarios, what does this technically mean?

  -   Will Data Holders reject Data Recipient's consumer data requests? What will the response code or error be in these scenarios? Is there an expectation that the DSB to explicitly define a consistent standard to guide participants? Or is a `403` HTTP status code and an `errors` object (left to the participant's discretion to populate) sufficient?
  -   Will Data Recipients be required to stop making (customer attended and unattended) requests to the Data Holder's consumer data sharing APIs?
  - Will the CDR Register have status and/or outage endpoints, similar to the endpoints defined in the [CDS](https://consumerdatastandardsaustralia.github.io/standards/#consumer-data-standards-common-apis) that could be used to inform participants and their behaviour when the CDR Register is unavailable?

- When the Register cache has expired and the CDR Register is still unavailable, what expected treatment is to be applied when a consent is revoked from the Data Recipient or Data Holder? We expect that the participant should honour the revocation request / event, however, notification to the counter-participant via the revocation endpoint may not be technically possible based on the latest metadata retrieved once the CDR Register is available. 

- More broadly, what are the technical implications to the other CDS InfoSec endpoints and are the expected flow on impacts understood? E.g. Customers will be impacted as they will not be able to complete a consent authorisation for data sharing.

#### Certificate Authority Unavailable

When the certificate status has expired and the CA is still unavailable, what expected treatment is to be applied when a consent is revoked from the Data Recipient or Data Holder? We expect that the participant should honour the revocation request / event, however, notification to the counter-participant via the revocation endpoint may not be technically possible based on the latest certificate status once the CA becomes available.

#### Accredited Data Recipient Unavailable

- Will there be a technical recommendation made on the polling (including exponential back-off) strategy for Data Holder's to implement when calling the Data Recipient's revocation endpoint? 

- How long after outage recovery must the Data Recipient refresh their cache from the CDR Register?

#### Accredited Data Holder Unavailable
- Will there be a technical recommendation made on the polling (including exponential back-off) strategy for Data Recipient's to implement when calling the Data Holder's revocation endpoint? Could Data Recipients make use of the Data Holder's [Get Status](https://consumerdatastandardsaustralia.github.io/standards/#get-status) and [Get Outages](https://consumerdatastandardsaustralia.github.io/standards/#get-outages) Common APIs as part of the retry strategy?

- How long after outage recovery must the Data Holder refresh their cache from the CDR Register?

**Comment by perlboy** [comment]
Starting with a note here that the expected target SLA of Holders as published in the [NFR's DP21](https://github.com/ConsumerDataStandardsAustralia/standards/issues/21) is specified as ``99.5%	per	month``. Presumably the ACCC Register SLA should be considered in this context and this translates to a maximum of **_3 hrs 40 minutes_** per month.

As a central cog there is an argument that ACCC should be targeting higher. Biza would suggest at least ``99.9%`` which translates to **_43 minutes_** of acceptable ecosystem outage time per month.

**Comment by JamesMBligh** [comment]
I agree the Register should target a high degree of availability.  Architecturally, however, having the availability of the register as a requirement for the ecosystem to function seems unnecessarily risky and goes against good design principles such as designing for resilience and redundancy.

In light if this, and considering that critical revocations are unlikely to occur frequently, I would be proponent of the an outage of the registry being considered a period of frozen status quo.  By this I mean that, if the register or CA is down then any previously validated certificates, keys or metadata should be considered valid until the register becomes available again.

If that is combined with a high availability goal for the register this would likely constitute a low risk position.
